src/Command/CompletionCommand.php

```diff
@@ -129,6 +130,9 @@ protected function getCommands(Arguments $args, ConsoleIo $io): int
         // to detect true duplicates (plugin-prefixed alias pointing to same command)
         $commandClasses = [];
         foreach ($this->commands as $key => $value) {
+            if (is_subclass_of($value, CommandHiddenInterface::class)) {
```

Comment:
```suggestion
            if ($value instanceof CommandHiddenInterface) {
```

Same for all other similar checks.

Reply:
I was a bit confused why the instance of checks I proposed would not work :)

Reply:
type annotation for the $commands property is 
```php
     * @var array<string, \Cake\Console\CommandInterface|class-string<\Cake\Console\CommandInterface>>
```

So we can have class-strings and instanceof requires an object on the left side. 

- Either the annotation is faulty
- or `is_subclass_of` is a better fit which works with both class-name or instances ?

Reply:
Fair enough I guess then.
We dont need to nano optimize for CLI anyway.

Reply:
Sorry, my bad. I didn't realize `$value` can be a string.